python/paddle/fluid/dygraph/dygraph_to_static/partial_program.py

```diff
@@ -112,14 +112,14 @@ def __init__(self, main_program, inputs, outputs, parameters=None):
         self._outputs = NestSequence(outputs, need_check=True)
         self._params = parameters if parameters is not None else []
 
-        self._infer_program = self._verify_program(main_program)
-        self._train_program = self._append_backward_desc()
-        # Switch infer or train by train() and eval()
-        self._trace_program = None
+        main_program = self._verify_program(main_program)
+        self._infer_program = self._clone_for_test(main_program)
```

Comment:
why not use `Program.clone(for_test=True)`directly?

Reply:
Thanks, done